src/main/java/org/terasology/launcher/game/GameManager.java

```diff
@@ -150,7 +151,10 @@ private void scanInstallationDir() {
                     }
                     for (File versionDirectory : Objects.requireNonNull(buildDirectory.listFiles())) {
                         String version = versionDirectory.getName();
-                        localGames.add(new GameIdentifier(version, build, profile));
+                        // FIXME: Assumes version==engineVersion. Should we pull engineVersion from the installation's
+                        //   files instead of the name of its directory?
```

Comment:
Where is the version written in the installation files? 

There's a `VERSION` file, but the content does not look very promising...

```
Terasology - Version
====================

Jenkins:
  URL: http://jenkins.terasology.io/teraorg/job/Terasology/job/engine/job/develop/399/
  Build number: 399
Created at: 2021-03-12T21:35:42Z

http://terasology.org
https://github.com/MovingBlocks/Terasology
```

Reply:
The engine version is encoded in `libs/engine-4.3.0-SNAPSHOT.jar`, but parsing it from that also sounds horrible...

Reply:
Yeah, that VERSION file is more for humans than something intended to be machine-readable.

I checked, and I think we do currently do _not_ have any `terasology --version` command that shows version information.

A thing that is kind of gross, but at least has the advantage of being something unambiguous, is to extract the versionInfo.properties from _inside_ that `engine.jar`.

Reply:
Implemented. (I think. Needs some manual testing because I don't have integration tests involving actual release files.)

Reply:
Added a comment about this in my review.